src/main/java/ch/njol/skript/expressions/ExprDetonationTime.java

```diff
@@ -0,0 +1,139 @@
+package ch.njol.skript.expressions;
+
+import ch.njol.skript.classes.Changer;
+import ch.njol.skript.classes.Changer.ChangeMode;
+import ch.njol.skript.doc.Description;
+import ch.njol.skript.doc.Examples;
+import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.Since;
+import ch.njol.skript.expressions.base.SimplePropertyExpression;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.skript.util.Timespan;
+import ch.njol.util.Kleenean;
+import ch.njol.util.coll.CollectionUtils;
+import org.bukkit.entity.Creeper;
+import org.bukkit.entity.Entity;
+import org.bukkit.entity.TNTPrimed;
+import org.bukkit.event.Event;
+import org.jetbrains.annotations.Nullable;
+
+@Name("Detonation Time")
+@Description({
+	"Sets or gets the detonation time of an entity.",
+	"Supports Primed TNT and Creepers. For Creepers, you can also retrieve their maximum fuse time."
+})
+@Examples({
+	"set detonation time of {_tnt} to 4 seconds",
+	"set max detonation time of last spawned creeper to 3 seconds",
+	"broadcast detonation time of {_tnt}",
+})
+@Since("INSERT VERSION")
+public class ExprDetonationTime extends SimplePropertyExpression<Entity, Timespan> {
+
+	static {
+		register(ExprDetonationTime.class, Timespan.class, "[:max[imum]] detonation time", "entities");
```

Comment:
I think supporting `fuse time` as an option would be good

Reply:
im refactoring the whole thing to fuse duration